cmd/scollector/collectors/snmp.go

```diff
@@ -4,19 +4,43 @@ import (
 	"fmt"
 	"io"
 	"math/big"
+	"time"
 
 	"bosun.org/_third_party/github.com/mjibson/snmp"
+	"bosun.org/cmd/scollector/conf"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
 )
 
-func SNMP(community, host string) error {
-	if host == "" {
+var builtInSnmps = map[string]func(cfg conf.SNMP){"ifaces": SNMPIfaces, "cisco": SNMPCisco}
+
+func SNMP(cfg conf.SNMP, mibs map[string]conf.MIB) error {
+	if cfg.Host == "" {
 		return fmt.Errorf("empty SNMP hostname")
 	}
-	if community == "" {
+	if cfg.Community == "" {
 		return fmt.Errorf("empty SNMP community")
 	}
-	SNMPIfaces(community, host)
-	SNMPCisco(community, host)
+	if len(cfg.MIBs) == 0 {
+		cfg.MIBs = []string{"ifaces", "cisco"}
+	}
+	for _, m := range cfg.MIBs {
+		if f, ok := builtInSnmps[m]; ok {
```

Comment:
If a name from builtInSNPMs and mibs collides, we should prefer using mibs since that's what the user specified. Switch the order of these if blocks.